src/GitVersion.Core/MergeMessage.cs

```diff
@@ -98,7 +98,11 @@ private ReferenceName GetMergedBranchName(string mergedBranch)
 
         mergeMessage = null;
 
-        if (mergeCommit.IsMergeCommit)
+        var mergeMessageFormats = DefaultFormats.Union(
+                configuration.MergeMessageFormats
+                    .Select(n => new MergeMessageFormat(n.Key, n.Value)));
+
+        if (mergeCommit.IsMergeCommit || mergeMessageFormats.Any(format => format.Pattern.IsMatch(mergeCommit.Message)))
```

Comment:
Can we expose this as `Commit.HasMergeMessage(…)` instead?

```suggestion
        if (mergeCommit.IsMergeCommit || mergeCommit.HasMergeMessage(mergeMessageFormats))
``````